custom_components/tuya_local/devices/yym_805SW_aroma_nightlight.yaml

```diff
@@ -7,18 +8,114 @@ primary_entity:
     - id: 1
       name: switch
       type: boolean
+      mapping:
+        - dps_val: true
+          icon: mdi:scent
+          value: ON
+        - dps_val: false
+          icon: mdi:scent-off
+          value: OFF      
     - id: 2
       name: speed
       type: string
+      hidden: true
       mapping:
         - dps_val: large
           value: 100
+          icon: "mdi:sprinkler-variant"
```

Comment:
If you are setting the icon from two dps on the same entity, you need to define the icon_priority to get consistent behaviour.  In this case I do not see the point, because there is full coverage of the options on both dps, so only one will take effect (but without priorities, it will be random which).

Reply:
I have explained this in comment below.